testplugins/src/main/java/org/spongepowered/test/CopyPasta.java

```diff
@@ -225,23 +227,37 @@ public void onGamePreInitialization(GamePreInitializationEvent event) {
             .build(), "load");
     }
 
-    @Listener
-    public void onInteract(InteractBlockEvent.Secondary.MainHand event, @Root Player player) {
-        Optional<ItemStack> item = player.getItemInHand(HandTypes.MAIN_HAND);
-        if (item.isPresent() && item.get().getType().equals(ItemTypes.WOODEN_AXE) && event.getTargetBlock() != BlockSnapshot.NONE) {
-            get(player).setPos2(event.getTargetBlock().getPosition());
-            player.sendMessage(Text.of(TextColors.LIGHT_PURPLE, "Position 2 set to " + event.getTargetBlock().getPosition()));
-            event.setCancelled(true);
-        }
+    @Override
+    public void disable(CommandSource src) {
+        Sponge.getEventManager().unregisterListeners(this.plugin);
```

Comment:
Could this be streamlined/default implemented so only the plugins that have more than just the listeners will need to override?

Reply:
I would need another method to retrieve the class handling the listeners to default implement this. 
Is it worth adding more methods ?

Reply:
I'd say it's worth adding a method returning an object as "listener".